### ISSUE TITLE
Deep Bayesian Active Learning for Multiple Correct Outputs

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#1]
## [Deep Bayesian Active Learning for Multiple Correct Outputs](https://arxiv.org/abs/1912.01119)
*Typical active learning strategies are designed for tasks, such as classification, with the assumption that the output space is mutually exclusive. The assumption that these tasks always have exactly one correct answer has resulted in the creation of numerous uncertainty-based measurements, such as entropy and least confidence, which operate over a model's outputs. Unfortunately, many real-world vision tasks, like visual question answering and image captioning, have multiple correct answers, causing these measurements to overestimate uncertainty and sometimes perform worse than a random sampling baseline. In this paper, we propose a new paradigm that estimates uncertainty in the model's internal hidden space instead of the model's output space. We specifically study a manifestation of this problem for visual question answer generation (VQA), where the aim is not to classify the correct answer but to produce a natural language answer, given an image and a question. Our method overcomes the paraphrastic nature of language. It requires a semantic space that structures the model's output concepts and that enables the usage of techniques like dropout-based Bayesian uncertainty. We build a visual-semantic space that embeds paraphrases close together for any existing VQA model. We empirically show state-of-art active learning results on the task of VQA on two datasets, being 5 times more cost-efficient on Visual Genome and 3 times more cost-efficient on VQA 2.0.*

### Cite this paper
```bibtex
@misc{jedoui2019deep,
    title={Deep Bayesian Active Learning for Multiple Correct Outputs},
    author={Khaled Jedoui and Ranjay Krishna and Michael Bernstein and Li Fei-Fei},
    year={2019},
    eprint={1912.01119},
    archivePrefix={arXiv},
    primaryClass={cs.CV}
}
```

**Comment by juanmc2005** [comment]
The estimated uncertainty of a model's prediction is a popular metric used in active learning in order to select new training examples to annotate.
This paper begins by explaining that existing active learning methods for uncertainty estimation are not better than randomly annotating new examples when it comes to sequence generation models. In addition to this, deep learning models are not well fit to tasks in which many outputs are equally correct. This makes language generation very difficult.

The authors propose a new approach for Visual Question Answering based on bayesian uncertainty estimation in the network's latent space, which they call _visual-semantic_, as they are constructed using the semantic content of both the image and the question.

Their method consists of training an encoder of images with their captions into hidden representations. Then they optimize triplet loss to cluster similar images and captions, and cross entropy to reconstruct captions from their embeddings.
Once this visual-semantic space is trained, they train another encoder-decoder architecture with pairs (q, i) as input and answers a as output. For that they use a cross entropy reconstruction loss and a MSE in the latent space with visual-semantic encodings as ground truths.

Finally, the uncertainty estimation is done using a Monte-Carlo dropout method to generate multiple answers for a pair (q, i). This makes the neural network a stochastic model, allowing the use of variance as an uncertainty estimation. These answers are then projected to the visual-semantic space and the sum of variances across all dimensions is calculated to measure uncertainty.

![Proposed model](https://raw.githubusercontent.com/juanmc2005/phd-bib/master/jedoui_etal2019-model.png)

In the active learning scenario, the K examples with the highest uncertainties are the ones chosen for labeling.
